diff -ur networkmanagement-orig/backends/NetworkManager/busconnection.h networkmanagement-0.9/backends/NetworkManager/busconnection.h
--- networkmanagement-orig/backends/NetworkManager/busconnection.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/busconnection.h	2011-04-20 14:53:34.895397075 +0200
@@ -51,9 +51,9 @@
 class BusConnection : public QObject
 {
     Q_OBJECT
-    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManagerSettings.Connection")
+    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManager.Settings.Connection")
     // Can QtDbus handle multiple interfaces being provided by one object like this?
-    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManagerSettings.Connection.Secrets")
+    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManager.Settings.Connection.Secrets")
 
     public:
         /**
diff -ur networkmanagement-orig/backends/NetworkManager/exportedconnection.h networkmanagement-0.9/backends/NetworkManager/exportedconnection.h
--- networkmanagement-orig/backends/NetworkManager/exportedconnection.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/exportedconnection.h	2011-04-20 14:53:34.895397075 +0200
@@ -30,9 +30,9 @@
 class ConnectionAdaptor: public QDBusAbstractAdaptor
 {
     Q_OBJECT
-    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManagerSettings.Connection")
+    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManager.Settings.Connection")
     Q_CLASSINFO("D-Bus Introspection", ""
-"  <interface name=\"org.freedesktop.NetworkManagerSettings.Connection\">\n"
+"  <interface name=\"org.freedesktop.NetworkManager.Settings.Connection\">\n"
 "    <method name=\"Update\">\n"
 "      <annotation value=\"impl_exported_connection_update\" name=\"org.freedesktop.DBus.GLib.CSymbol\"/>\n"
 "      <annotation value=\"\" name=\"org.freedesktop.DBus.GLib.Async\"/>\n"
diff -ur networkmanagement-orig/backends/NetworkManager/exportedconnectionsecrets.h networkmanagement-0.9/backends/NetworkManager/exportedconnectionsecrets.h
--- networkmanagement-orig/backends/NetworkManager/exportedconnectionsecrets.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/exportedconnectionsecrets.h	2011-04-20 14:53:34.895397075 +0200
@@ -30,9 +30,9 @@
 class SecretsAdaptor: public QDBusAbstractAdaptor
 {
     Q_OBJECT
-    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManagerSettings.Connection.Secrets")
+    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManager.Settings.Connection.Secrets")
     Q_CLASSINFO("D-Bus Introspection", ""
-"  <interface name=\"org.freedesktop.NetworkManagerSettings.Connection.Secrets\">\n"
+"  <interface name=\"org.freedesktop.NetworkManager.Settings.Connection.Secrets\">\n"
 "    <method name=\"GetSecrets\">\n"
 "      <annotation value=\"impl_exported_connection_get_secrets\" name=\"org.freedesktop.DBus.GLib.CSymbol\"/>\n"
 "      <annotation value=\"\" name=\"org.freedesktop.DBus.GLib.Async\"/>\n"
diff -ur networkmanagement-orig/backends/NetworkManager/introspection/nm-active-connection.xml networkmanagement-0.9/backends/NetworkManager/introspection/nm-active-connection.xml
--- networkmanagement-orig/backends/NetworkManager/introspection/nm-active-connection.xml	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/introspection/nm-active-connection.xml	2011-04-20 14:53:34.895397075 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 
 <node name="/" xmlns:tp="http://telepathy.freedesktop.org/wiki/DbusSpec#extensions-v0">
-  <interface name="org.freedesktop.NetworkManager.Connection.Active">
+  <interface name="org.freedesktop.NetworkManagerCompat.Connection.Active">
     <property name="ServiceName" type="s" access="read">
       <tp:docstring>The D-Bus service name providing this connection.</tp:docstring>
     </property>
diff -ur networkmanagement-orig/backends/NetworkManager/introspection/nm-connection-secrets.xml networkmanagement-0.9/backends/NetworkManager/introspection/nm-connection-secrets.xml
--- networkmanagement-orig/backends/NetworkManager/introspection/nm-connection-secrets.xml	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/introspection/nm-connection-secrets.xml	2011-04-20 14:53:34.895397075 +0200
@@ -2,7 +2,7 @@
 
 <node name="/" xmlns:tp="http://telepathy.freedesktop.org/wiki/DbusSpec#extensions-v0">
 
-    <interface name="org.freedesktop.NetworkManagerSettings.Connection.Secrets">
+    <interface name="org.freedesktop.NetworkManager.Settings.Connection.Secrets">
         <tp:docstring>
             Secrets have a separate interface so that they can be locked down.
         </tp:docstring>
diff -ur networkmanagement-orig/backends/NetworkManager/introspection/nm-exported-connection.xml networkmanagement-0.9/backends/NetworkManager/introspection/nm-exported-connection.xml
--- networkmanagement-orig/backends/NetworkManager/introspection/nm-exported-connection.xml	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/introspection/nm-exported-connection.xml	2011-04-20 14:53:34.895397075 +0200
@@ -2,7 +2,7 @@
 
 <node name="/" xmlns:tp="http://telepathy.freedesktop.org/wiki/DbusSpec#extensions-v0">
 
-    <interface name="org.freedesktop.NetworkManagerSettings.Connection">
+    <interface name="org.freedesktop.NetworkManager.Settings.Connection">
         <tp:docstring>
             Represents a single network connection configuration.
         </tp:docstring>
diff -ur networkmanagement-orig/backends/NetworkManager/introspection/nm-settings.xml networkmanagement-0.9/backends/NetworkManager/introspection/nm-settings.xml
--- networkmanagement-orig/backends/NetworkManager/introspection/nm-settings.xml	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/introspection/nm-settings.xml	2011-04-20 14:53:34.896397057 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 
 <node name="/" xmlns:tp="http://telepathy.freedesktop.org/wiki/DbusSpec#extensions-v0">
-  <interface name="org.freedesktop.NetworkManagerSettings">
+  <interface name="org.freedesktop.NetworkManager.Settings">
     <tp:docstring>
       The NetworkManagerSettings interface is provided by the service which provides connections to NetworkManager.
     </tp:docstring>
diff -ur networkmanagement-orig/backends/NetworkManager/introspection/nm-vpn-connection.xml networkmanagement-0.9/backends/NetworkManager/introspection/nm-vpn-connection.xml
--- networkmanagement-orig/backends/NetworkManager/introspection/nm-vpn-connection.xml	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/introspection/nm-vpn-connection.xml	2011-04-20 14:53:34.896397057 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 
 <node name="/" xmlns:tp="http://telepathy.freedesktop.org/wiki/DbusSpec#extensions-v0">
-  <interface name="org.freedesktop.NetworkManager.VPN.Connection">
+  <interface name="org.freedesktop.NetworkManagerCompat.VPN.Connection">
     <tp:docstring>
       Represents an active connection to a Virtual Private Network.
     </tp:docstring>
diff -ur networkmanagement-orig/backends/NetworkManager/nm-active-connectioninterface.h networkmanagement-0.9/backends/NetworkManager/nm-active-connectioninterface.h
--- networkmanagement-orig/backends/NetworkManager/nm-active-connectioninterface.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/nm-active-connectioninterface.h	2011-04-20 16:49:16.302308474 +0200
@@ -29,7 +29,7 @@
     Q_OBJECT
 public:
     static inline const char *staticInterfaceName()
-    { return "org.freedesktop.NetworkManager.Connection.Active"; }
+    { return "org.freedesktop.NetworkManagerCompat.Connection.Active"; }
 
 public:
     OrgFreedesktopNetworkManagerConnectionActiveInterface(const QString &service, const QString &path, const QDBusConnection &connection, QObject *parent = 0);
diff -ur networkmanagement-orig/backends/NetworkManager/nmdbussettingsservice.cpp networkmanagement-0.9/backends/NetworkManager/nmdbussettingsservice.cpp
--- networkmanagement-orig/backends/NetworkManager/nmdbussettingsservice.cpp	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/nmdbussettingsservice.cpp	2011-04-20 14:53:34.894397093 +0200
@@ -57,6 +57,9 @@
     QHash<QUuid, QDBusObjectPath> uuidToPath;
 };
 
+#define NM_DBUS_SERVICE_USER_SETTINGS     "org.freedesktop.NetworkManagerUserSettings"
+#define NM_DBUS_SERVICE_SYSTEM_SETTINGS   "org.freedesktop.NetworkManagerSystemSettings"
+
 const QString NMDBusSettingsService::SERVICE_USER_SETTINGS = QLatin1String(NM_DBUS_SERVICE_USER_SETTINGS);
 const QString NMDBusSettingsService::SERVICE_SYSTEM_SETTINGS = QLatin1String(NM_DBUS_SERVICE_SYSTEM_SETTINGS);
 
diff -ur networkmanagement-orig/backends/NetworkManager/nmdbussettingsservice.h networkmanagement-0.9/backends/NetworkManager/nmdbussettingsservice.h
--- networkmanagement-orig/backends/NetworkManager/nmdbussettingsservice.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/nmdbussettingsservice.h	2011-04-20 14:53:34.896397057 +0200
@@ -43,7 +43,7 @@
 class KNM_EXPORT NMDBusSettingsService : public QObject, public ActivatableObserver, public ConnectionHandler
 {
 Q_OBJECT
-Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManagerSettings")
+Q_CLASSINFO("D-Bus Interface", "org.freedesktop.NetworkManager.Settings")
 
 public:
     enum ServiceStatus { Available, AlreadyRunning, AccessDenied, UnknownError };
diff -ur networkmanagement-orig/backends/NetworkManager/nm-exported-connectioninterface.h networkmanagement-0.9/backends/NetworkManager/nm-exported-connectioninterface.h
--- networkmanagement-orig/backends/NetworkManager/nm-exported-connectioninterface.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/nm-exported-connectioninterface.h	2011-04-20 14:53:34.896397057 +0200
@@ -29,7 +29,7 @@
     Q_OBJECT
 public:
     static inline const char *staticInterfaceName()
-    { return "org.freedesktop.NetworkManagerSettings.Connection"; }
+    { return "org.freedesktop.NetworkManager.Settings.Connection"; }
 
 public:
     OrgFreedesktopNetworkManagerSettingsConnectionInterface(const QString &service, const QString &path, const QDBusConnection &connection, QObject *parent = 0);
diff -ur networkmanagement-orig/backends/NetworkManager/nm-exported-connection-secrets-interface.h networkmanagement-0.9/backends/NetworkManager/nm-exported-connection-secrets-interface.h
--- networkmanagement-orig/backends/NetworkManager/nm-exported-connection-secrets-interface.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/nm-exported-connection-secrets-interface.h	2011-04-20 16:49:56.551556038 +0200
@@ -29,7 +29,7 @@
     Q_OBJECT
 public:
     static inline const char *staticInterfaceName()
-    { return "org.freedesktop.NetworkManagerSettings.Connection.Secrets"; }
+    { return "org.freedesktop.NetworkManager.Settings.Connection.Secrets"; }
 
 public:
     OrgFreedesktopNetworkManagerSettingsConnectionSecretsInterface(const QString &service, const QString &path, const QDBusConnection &connection, QObject *parent = 0);
diff -ur networkmanagement-orig/backends/NetworkManager/nm-settingsinterface.h networkmanagement-0.9/backends/NetworkManager/nm-settingsinterface.h
--- networkmanagement-orig/backends/NetworkManager/nm-settingsinterface.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/nm-settingsinterface.h	2011-04-20 14:53:34.896397057 +0200
@@ -29,7 +29,7 @@
     Q_OBJECT
 public:
     static inline const char *staticInterfaceName()
-    { return "org.freedesktop.NetworkManagerSettings"; }
+    { return "org.freedesktop.NetworkManager.Settings"; }
 
 public:
     OrgFreedesktopNetworkManagerSettingsInterface(const QString &service, const QString &path, const QDBusConnection &connection, QObject *parent = 0);
diff -ur networkmanagement-orig/backends/NetworkManager/nm-vpn-connectioninterface.h networkmanagement-0.9/backends/NetworkManager/nm-vpn-connectioninterface.h
--- networkmanagement-orig/backends/NetworkManager/nm-vpn-connectioninterface.h	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/nm-vpn-connectioninterface.h	2011-04-20 16:52:07.726103441 +0200
@@ -28,7 +28,7 @@
     Q_OBJECT
 public:
     static inline const char *staticInterfaceName()
-    { return "org.freedesktop.NetworkManager.VPN.Connection"; }
+    { return "org.freedesktop.NetworkManagerCompat.VPN.Connection"; }
 
 public:
     OrgFreedesktopNetworkManagerVPNConnectionInterface(const QString &service, const QString &path, const QDBusConnection &connection, QObject *parent = 0);
diff -ur networkmanagement-orig/backends/NetworkManager/settings/gsmdbus.cpp networkmanagement-0.9/backends/NetworkManager/settings/gsmdbus.cpp
--- networkmanagement-orig/backends/NetworkManager/settings/gsmdbus.cpp	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/backends/NetworkManager/settings/gsmdbus.cpp	2011-04-20 14:53:34.894397093 +0200
@@ -35,7 +35,7 @@
   if (map.contains(QLatin1String(NM_SETTING_GSM_NETWORK_TYPE))) {
     setting->setNetworktype(map.value(QLatin1String(NM_SETTING_GSM_NETWORK_TYPE)).value<int>());
   } else {
-      setting->setNetworktype(NM_GSM_NETWORK_ANY);
+      setting->setNetworktype(NM_SETTING_GSM_NETWORK_TYPE_ANY);
   }
   if (map.contains("band")) {
     setting->setBand(map.value("band").value<int>());
@@ -58,7 +58,7 @@
       map.insert("apn", setting->apn());
   if (!setting->networkid().isEmpty())
       map.insert(QLatin1String(NM_SETTING_GSM_NETWORK_ID), setting->networkid());
-  if (setting->networktype() != NM_GSM_NETWORK_ANY) {
+  if (setting->networktype() != NM_SETTING_GSM_NETWORK_TYPE_ANY) {
       map.insert(QLatin1String(NM_SETTING_GSM_NETWORK_TYPE), setting->networktype());
   }
 
Pouze v networkmanagement-0.9/backends/NetworkManager/settings: gsmdbus.cpp.orig
diff -ur networkmanagement-orig/NetworkManager-kde4.conf networkmanagement-0.9/NetworkManager-kde4.conf
--- networkmanagement-orig/NetworkManager-kde4.conf	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/NetworkManager-kde4.conf	2011-04-20 14:53:34.896397057 +0200
@@ -14,23 +14,23 @@
                 <allow own="org.freedesktop.NetworkManagerUserSettings"/>
 
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
-                       send_interface="org.freedesktop.NetworkManagerSettings"/>
+                       send_interface="org.freedesktop.NetworkManager.Settings"/>
 
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
-                       send_interface="org.freedesktop.NetworkManagerSettings.Connection"/>
+                       send_interface="org.freedesktop.NetworkManager.Settings.Connection"/>
 
                 <!-- Only root can get secrets -->
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
-                       send_interface="org.freedesktop.NetworkManagerSettings.Connection.Secrets"/>
+                       send_interface="org.freedesktop.NetworkManager.Settings.Connection.Secrets"/>
         </policy>
         <policy at_console="true">
                 <allow own="org.freedesktop.NetworkManagerUserSettings"/>
 
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
-                       send_interface="org.freedesktop.NetworkManagerSettings"/>
+                       send_interface="org.freedesktop.NetworkManager.Settings"/>
 
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
-                       send_interface="org.freedesktop.NetworkManagerSettings.Connection"/>
+                       send_interface="org.freedesktop.NetworkManager.Settings.Connection"/>
         </policy>
         <policy context="default">
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
diff -ur networkmanagement-orig/settings/config/manageconnectionwidget.cpp networkmanagement-0.9/settings/config/manageconnectionwidget.cpp
--- networkmanagement-orig/settings/config/manageconnectionwidget.cpp	2011-04-20 13:36:45.000000000 +0200
+++ networkmanagement-0.9/settings/config/manageconnectionwidget.cpp	2011-04-20 14:53:34.894397093 +0200
@@ -59,6 +59,9 @@
 #define ConnectionLastUsedRole Qt::UserRole + 3
 #define ConnectionScopeRole Qt::UserRole + 4
 
+#define NM_DBUS_SERVICE_USER_SETTINGS     "org.freedesktop.NetworkManagerUserSettings"
+#define NM_DBUS_SERVICE_SYSTEM_SETTINGS   "org.freedesktop.NetworkManagerSystemSettings"
+
 K_PLUGIN_FACTORY( ManageConnectionWidgetFactory, registerPlugin<ManageConnectionWidget>();)
 K_EXPORT_PLUGIN( ManageConnectionWidgetFactory( "kcm_networkmanagement", "libknetworkmanager" ) )
